### PR TITLE
Add autofocus support

### PR DESCRIPTION
**Pull request by corsonr**
Add autofocus support to the WC fields, and set the autofocus to the 1st field on the checkout page on the frontend, that's to say the Billing First Name field.

The autofocus attribute can be widely removed using a filter.

### CODE REVIEW

File: includes/wc-template-functions.php [hunk 1]
```diff
@@ -1877,6 +1878,8 @@ function woocommerce_form_field( $key, $args, $value = null ) {
 		$args['maxlength'] = ( $args['maxlength'] ) ? 'maxlength="' . absint( $args['maxlength'] ) . '"' : '';
 
 		$args['autocomplete'] = ( $args['autocomplete'] ) ? 'autocomplete="' . esc_attr( $args['autocomplete'] ) . '"' : '';
+		
+		$args['autofocus'] = ( true == esc_attr( $args['autofocus'] ) ) ? apply_filters( 'woocommerce_form_field_autofocus', ' autofocus' ) : '' ;
```
Comment (mikejolley) [inline includes/wc-template-functions.php]: esc_attr should not be used for comparison because it is used for sanitizing output.

Comment (mikejolley) [inline includes/wc-template-functions.php]: True should also be a literal comparison (`===`).

Comment (mikejolley) [inline includes/wc-template-functions.php]: There is an extra space before `' autofocus'`

File: includes/wc-template-functions.php [hunk 2]
```diff
@@ -1980,7 +1983,7 @@ function woocommerce_form_field( $key, $args, $value = null ) {
 			case 'tel' :
 			case 'number' :
 
-				$field .= '<input type="' . esc_attr( $args['type'] ) . '" class="input-text ' . esc_attr( implode( ' ', $args['input_class'] ) ) . '" name="' . esc_attr( $key ) . '" id="' . esc_attr( $args['id'] ) . '" placeholder="' . esc_attr( $args['placeholder'] ) . '" ' . $args['maxlength'] . ' ' . $args['autocomplete'] . ' value="' . esc_attr( $value ) . '" ' . implode( ' ', $custom_attributes ) . ' />';
+				$field .= '<input type="' . esc_attr( $args['type'] ) . '" class="input-text ' . esc_attr( implode( ' ', $args['input_class'] ) ) . '" name="' . esc_attr( $key ) . '" id="' . esc_attr( $args['id'] ) . '" placeholder="' . esc_attr( $args['placeholder'] ) . '" ' . $args['maxlength'] . ' ' . $args['autocomplete'] . ' value="' . esc_attr( $value ) . '" ' . implode( ' ', $custom_attributes ) . $args['autofocus'] . ' />';
```
Comment (mikejolley) [inline includes/wc-template-functions.php]: This needs an escape.

**Review by mikejolley — CHANGES_REQUESTED** [review]
Just needs a few fixes before merge.

File: includes/wc-template-functions.php (hunk 1 above)
Comment (corsonr) [inline includes/wc-template-functions.php]: Actually it was on purpose as it needs a space when added to a form tag, or should I add the space directly inside each form tag?

**Review by claudiosanches — APPROVED** [review]
Ok this one

**Review by claudiosanches — CHANGES_REQUESTED** [review]
It's ok, but should go with the `$custom_attributes`, since it's more easy and need less code to do it.

After `$custom_attributes = array();` you can add:

``` php
if ( true === $args['autofocus'] ) {
    $custom_attributes['autofocus'] = 'autofocus';
}
```

This make works with all fields and already escape all things.

**Pull request merged**